puts/vendor/configs/openssl.toml

```diff
@@ -0,0 +1,52 @@
+[preset]
+default = "openssl312"
+
+[preset.openssl312]
+name = "openssl312"
+fetch = "openssl312"
+build = ["base", "parallel"]
+
+[preset.openssl111k]
+name = "openssl111k"
+fetch = "openssl111k"
+build = ["base", "parallel"]
+
+[preset.openssl111j]
+name = "openssl111j"
+fetch = "openssl111j"
+build = ["base", "parallel"]
+
+[preset.openssl111u]
+name = "openssl111u"
+fetch = "openssl111u"
+build = ["base", "parallel"]
+
+[preset.openssl101f]
+name = "openssl101f"
+fetch = "openssl101f"
+build = ["base"]
+
+[preset.openssl102u]
+name = "openssl102u"
+fetch = "openssl102u"
+build = ["base"]
+
+[fetch]
+openssl312 = { git = "https://github.com/tlspuffin/openssl", ref = "fuzz-OpenSSL_3_1_2" }
+openssl111k = { git = "https://github.com/tlspuffin/openssl", ref = "fuzz-OpenSSL_1_1_1k" }
+openssl111j = { git = "https://github.com/tlspuffin/openssl", ref = "fuzz-OpenSSL_1_1_1j" }
+openssl111u = { git = "https://github.com/tlspuffin/openssl", ref = "fuzz-OpenSSL_1_1_1u" }
+openssl101f = { git = "https://github.com/tlspuffin/openssl", ref = "OpenSSL_1_0_1f" }
+openssl102u = { git = "https://github.com/tlspuffin/openssl", ref = "OpenSSL_1_0_2u" }
+
+[patch]
+
+[build.base]
+cmake = "openssl"
+options = ["asan", "sancov", "gcov", "llvm_cov"]
```

Comment:
Don't we want to have asan as opt-in? With something like `[build.asan]` and then we can pick up a preset and add `asan` when we want to compile this preset but with Asan enabled.

Reply:
The `options` in the `.toml` declare the options but they are not set by default. You add `--options=asan` to build a given preset with asan, e.g.:
```sh
$ mk_vendor openssl:openssl111k --options=sancov,asan
```

So I think this is working as you expect. Or am I not understanding what you are asking for?

Reply:
RIght! I got confused ;)